src/pytanis/pretalx/client.py

```
@@ -110,96 +201,141 @@ def _endpoint_id(
         """Lists all answers and their details"""
         return self._endpoint_lst(Answer, event_slug, 'answers', params=params)
 
-    def tag(self, event_slug: str, tag: str, *, params: QueryParams | None = None) -> Tag:
+    def tag(self, event_slug: str, tag: str, *, params: QueryParams | dict | None = None) -> Tag:
         """Returns a specific tag"""
         return self._endpoint_id(Tag, event_slug, 'tags', tag, params=params)
 
-    def tags(self, event_slug: str, *, params: QueryParams | None = None) -> tuple[int, Iterator[Tag]]:
+    def tags(self, event_slug: str, *, params: QueryParams | dict | None = None) -> tuple[int, Iterator[Tag]]:
         """Lists all tags and their details"""
         return self._endpoint_lst(Tag, event_slug, 'tags', params=params)
 
+    def submission_type(self, event_slug: str, id: int, *, params: QueryParams | dict | None = None) -> SubmissionType:  # noqa: A002
+        """Returns a specific submission type"""
+        return self._endpoint_id(SubmissionType, event_slug, 'submission-types', id, params=params)
+
+    def submission_types(
+        self, event_slug: str, *, params: QueryParams | dict | None = None
+    ) -> tuple[int, Iterator[SubmissionType]]:
+        """Lists all submission types and their details"""
+        return self._endpoint_lst(SubmissionType, event_slug, 'submission-types', params=params)
+
+    def track(self, event_slug: str, id: int, *, params: QueryParams | dict | None = None) -> Track:  # noqa: A002
+        """Returns a specific track"""
+        return self._endpoint_id(Track, event_slug, 'tracks', id, params=params)
+
+    def tracks(self, event_slug: str, *, params: QueryParams | dict | None = None) -> tuple[int, Iterator[Track]]:
+        """Lists all tracks and their details"""
+        return self._endpoint_lst(Track, event_slug, 'tracks', params=params)
+
+    @classmethod
+    def __validate(cls, model_type, result):
```

- what does this method do? It more or less replaces one call with another. Instead of model_type.model_validate(result), you type __valide(model_type, result) with the difference that validation errors of pydantic are just swallowed/ignored?

There are also pydantic settings for instance to allow new fields without erroring that could do the same.

---

You’re right to question that. The abstraction wraps four different pydantic calls because the pretix API isn’t always consistent—some validations failed despite the data being mostly correct. 
This wrapper makes it easier to identify the root cause. 

Swallowing validation errors is **debatable**, but the real question is **how we want to handle partial failures**. 
For example, if the API returns 444 items but only 443 validate—should we:
- drop everything or 
- allow partial results? 
I’d prefer giving users an explicit choice in such cases.